File: Basic/5.Check if a number is a palindrome/Solution.cpp
#include <iostream>
using namespace std;

// Function to check if a number is a palindrome
bool isPalindrome(int num) {
    int original = num, reversed = 0, remainder;

    while (num != 0) {
        remainder = num % 10;
        reversed = reversed * 10 + remainder;
        num /= 10;
    }

    return original == reversed; // Return true if original equals reversed
}

int main() {
    int num;

    cout << "Enter a number: ";
    cin >> num;

    if (isPalindrome(num)) {
        cout << "The number is a palindrome." << endl;
    } else {
        cout << "The number is not a palindrome." << endl;
    }

    return 0;
}

File: Basic/5.Check if a number is a palindrome/Solution.py
# Function to check if a number is a palindrome
def is_palindrome(num):
    original = num
    reversed_num = 0

    while num != 0:
        remainder = num % 10
        reversed_num = reversed_num * 10 + remainder
        num //= 10

    return original == reversed_num  # Return true if original equals reversed

# Input from the user
num = int(input("Enter a number: "))

if is_palindrome(num):
    print("The number is a palindrome.")
else:
    print("The number is not a palindrome.")
